src/execution/physical_plan/plan_create_index.cpp

```diff
@@ -12,9 +14,26 @@ namespace duckdb {
 
 unique_ptr<PhysicalOperator> PhysicalPlanGenerator::CreatePlan(LogicalCreateIndex &op) {
 
+	// generate a physical plan for the parallel index creation which consists of the following operators
+	// table scan - projection (for expression execution) - filter (NOT NULL) - order - create index
+
 	D_ASSERT(op.children.empty());
 
+	// validate that all expressions contain valid scalar functions
+	// e.g. get_current_timestamp(), random(), and sequence values are not allowed as ART keys
+	// because they make deletions and lookups unfeasible
+	for (idx_t i = 0; i < op.unbound_expressions.size(); i++) {
+		auto &expr = op.unbound_expressions[i];
+		if (expr->expression_class == ExpressionClass::BOUND_FUNCTION) {
```

Comment:
Can we use `expr->HasSideEffects()` here? That also checks this property recursively in the children of the expression.

Reply:
Sure, I added a reminder to #5865.